File: beware-of-the-sheep/src/Entities/Plants/Belladonna.cpp
#include "pch.h"
#include "Belladonna.h"
#include "World.h"
#include "Entities/Animals/Animal.h"

namespace bots {

	Belladona::Belladona(World & world, const Point & position) :
		Plant(world, position, 99)
	{}

	void Belladona::collision(Entity & other)
	{
		if (dynamic_cast<Animal *>(&other)) {
			if (other.getStrength() < getStrength())
				other.kill();
			std::string message = other.getClassName() + " was slain by " + getClassName();
			m_world.addMessage(std::move(message));
		}
	}

} // namespace bots

File: src/com/kostu96/bots/entities/plants/Belladonna.java
package com.kostu96.bots.entities.plants;

import com.kostu96.bots.World;
import com.kostu96.bots.entities.Entity;
import com.kostu96.bots.entities.animals.Animal;
import com.kostu96.bots.utils.ImageManager.ImageID;
import com.kostu96.bots.utils.Point;

public class Belladonna extends Plant {
	private static final int STRENGTH = 99;

	public Belladonna(World world, Point position) {
		super(world, position, STRENGTH);
	}

	@Override
	public void collision(Entity other) {
		if (other instanceof Animal && other.getStrength() < getStrength()) {
			other.kill();
			
			// TODO: messages
			System.out.println(other.getClassName() + " was slain by " + getClassName());
		}
	}

	@Override
	public ImageID getImageID() {
		return ImageID.BELLADONA;
	}

}
